---
title: "Characterizing Evolutional Processes in Animals"
author: "Tony Ianniello"
bibliography: references.bib
date: '2022-04-27'
output:
  html_document: default
  pdf_document: default
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

## Opening Background

Generally, there are three main evolutionary processes that all organisms undergo. In stabilizing selection, the range of a population's phenotypes narrow on a small range. Arctic reindeer have experienced this type of selection on the mass of their calf at birth [@holand2019]. Under directional selection, one extreme phenotype is selected for and the opposite is selected against. In Galapagos finches, directional selection has acted on beak lengths, as the birds have had longer beaks over a 40 year period [@daphnef]. The last form of selection, divergent selection, occurs when the extreme values are selected for and the average value is unfit. The mating call of poison dart frogs in the Andes has experienced divergent selection, as higher rates are preferred in the south and lower rates are preferred in the north [@brusa2012].

New methods of testing for evolution are published every year. More recently, developments in genetics and DNA analysis has allowed for more precise estimations of evolutionary processes in subsets of populations. Many of these methods allow for phylogenetic analysis of long-deceased organisms through the fossil record [@castiglione2018]. In one study, scientists used genetic manipulations to engineer fruit flies with reconstructed ancient genes in order to determine how evolution affected those genes [@siddiq2017].

## Library Loading

```{r install packages, message = FALSE}
# install.packages(c("ggplot2","tidyverse","explore","gt"))
library(ggplot2)
library(tidyverse)
library(explore)
library(gt)
```

## Data Sources

#### Ungulate Dataset

Arctic reindeer data was accessed from Dryad [@holand2019].

```{r Ungulate Dataset, message = FALSE}
ungulate_selection <- read_csv("natural_selection_data.csv")
ungulate_pedigree <- read_csv("pedigree_data.csv")
ungulate_phenotypes <- read_csv("phenotypic_change_data.csv")
reindeer <- read_csv("reindeer_2019-11-19.csv")
```

#### Finch Dataset

Finch data was accessed from Dryad [@daphnef].

```{r Finch Dataset, message = FALSE}
beaks1975 <- read_csv("finch_beaks_1975.csv")
beaks2012 <- read_csv("finch_beaks_2012.csv")
fortis <- read_csv("fortis_beak_depth_heredity.csv")
scandens <- read_csv("scandens_beak_depth_heredity.csv")
fortis_time <- read_csv("fortis_time.csv")
scandens_time <- read_csv("scandens_time.csv",
                         col_types = cols("CI Beak length" = col_number(),
                                          "CI Beak depth" = col_number(),
                                          "CI beak width" = col_number()))

beaks1975 %>%
  rename(beak_length = "Beak length, mm", beak_depth = "Beak depth, mm") %>%
  add_column(year = 1975) -> beaks1975
beaks2012 %>%
  rename(beak_length = "blength", beak_depth = "bdepth") %>%
  add_column(year = 2012) -> beaks2012
beak_data <- full_join(beaks1975,beaks2012)
fortis %>%
  mutate(species = "fortis") %>%
  relocate(species, mid_offspring = `Mid-offspr`) -> fortis
scandens %>%
  mutate(species = "scandens") %>%
  relocate(species, mid_offspring) -> scandens
scandens_time %>%
  rename("CI Beak width" = "CI beak width") %>%
  full_join(fortis_time) %>%
  rename(beak_length = "Beak length",
         beak_width = "Beak width",
         beak_depth = "Beak depth") %>%
  na.omit() -> finch_time
```

#### Poison Dart Frog Dataset

Poison dart frog data was accessed directly from an article in the Journal of Biogeography [@brusa2012].

```{r Poison Dart Frog Dataset, message = FALSE}
frogs1 <- select(read_csv("frogs1.csv"),-(7:19))
frogs2 <- select(read_csv("frogs2.csv"),-(10:11))
frogs <- full_join(frogs1,frogs2)[-c(10:11),]
names(frogs) <- c("Population","n","PR","CD","CR","DF","Code","Region","Cyt_b","16S","Ms","Color","Call","Body_size")
```

## Figures

#### Figure 1

The purpose of Figure 1 is to demonstrate stabilizing selection in arctic reindeer birthmass over time. This figure shows that birth mass had been increasing linearly for arctic ungulates until 2005, when the value became constant. I chose to include a best fit curve rather than a line because the data does not fit well in a linear best fit and forcing it into one would not show the stabilizing selection that began around 2005. I also removed all gridlines to focus the viewer on the key data trend rather than the individual data points, which are only important in the context of the other data.

```{r Stabilizing Selection w/ Ungulates, message=FALSE}
reindeer %>%
  # group by year and only include data since 1990
  group_by(year) %>%
  filter(year >= 1990) %>%
  # make summary table with average brith mass and other variables
  summarise(n = n(),
            birthdate_mean = mean(birth_date), birthdate_sd = sd(birth_date),
            birthmass_mean = mean(birth_mass), birthmass_sd = sd(birth_mass)) %>%
  ggplot(aes(x = year, y = birthmass_mean)) +
    # add a point geometry comparing birthmass over time
    geom_point(color = "#377eb8") +
    # add a best fit curve with error
    geom_smooth(fill = "orange", color = "#e41a1c") +
    # set labels for title, caption, and x and y axes
    labs(title = "Stabilizing Selection\nin Arctic Ungulates",
         caption = "Figure 1. Birth mass of arctic ungulates from 1990 to 2015.",
         x = "Year", y = "Mass at Birth (kg)") +
    # set theme to black & white to match the other figures
    theme_bw() +
    theme(# set all text font to Bookman
          text = element_text(family = "Bookman"),
          # change sizing and location of title and caption
          plot.title = element_text(hjust = 0.5, size = 15),
          plot.caption = element_text(hjust = 0, size = 10),
          # remove the gridlines to make figure simpler
          panel.grid = element_blank())

```

#### Figure 2

The purpose of Figure 2 is to demonstrate directional selection in Galapagos finch beak lengths over time. I included the first figure in my infographic, which shows the distribution of beak lengths for finches on one island. I wanted to include two histograms to demonstrate the changes in distributions because populations are made of many different individuals, and I included a single vertical line to explicitly show the viewer where the mean value lies for each year. That, along with the x axis, clearly shows that beak lengths have increased over time. In the second figure on finches, I tracked the depth of beaks for two different species for each year since 1975. G. fortis beak depths have decreased dramatically over that period, while G. scandens beaks have remained relatively constant. Similarly to Figure 1, I chose to remove all gridlines and focus the viewer on the patterns of data I wanted them to see. I fit the data for both species to a linear best fit because the pattern is fairly constant for the period.

```{r Directional Selection w/ Finch Data, message=FALSE}
beak_data %>%
  # group by year and summarise data to include the average beak length
  group_by(year) %>%
  summarise(avg = mean(beak_length)) %>%
  # do a full join with beak_data to have average beak length within the original
  # dataframe
  full_join(beak_data) %>%
  ggplot() +
    # make a histogram geometry showing beak length distribution
    geom_histogram(aes(x = beak_length), binwidth = 0.2,
                   alpha = 0.3, fill = "#377eb8", color = "black") +
    # make vertical line for average 
    geom_vline(aes(xintercept = avg), color = "#e41a1c",
               linetype = "dashed") +
    # make facets for two different years in dataset (1975, 2012)
    facet_wrap(vars(year)) +
    # set labels for title, caption, and x and y axes
    labs(title = "Directional Selection in Finches",
         caption = "Figure 2. Beak length of two species of finch from 1975 to 2012.",
         x = "Beak Length", y = "Count") +
    # set theme to black & white to match the other figures 
    theme_bw() +
    theme(# set all text font to Bookman
          text = element_text(family = "Bookman"),
          # change sizing and location of title and caption
          plot.title = element_text(hjust = 0.5, size = 15),
          plot.caption = element_text(hjust = 0, size = 10),
          # remove the gridlines to make figure simpler
          panel.grid = element_blank())

ggplot(finch_time, aes(x = Year, y = beak_depth, color = Species)) +
  # make a point geometry comparying beak lengths over time
  geom_point() +
  # add a line of best fit with error
  geom_smooth(method = "lm") +
  # set labels for caption, title, and y axis
  labs(y = "Beak Depth (mm)",
       title = "Directional Selection in Finches",
       caption = "Figure 2. Beak length of two species of finch from 1975 to 2012.") +
  # set color values to match Set 1 in ColorBrewer
  scale_color_manual(values = c("#e41a1c","#377eb8")) +
  # set theme to black & white to match the other figures
  theme_bw() +
  theme(# set all text font to Bookman
        text = element_text(family = "Bookman"),
        # change sizing and location of title and caption
        plot.title = element_text(hjust = 0.5, size = 15),
        plot.caption = element_text(hjust = 0, size = 10),
        # remove the gridlines to make figure simpler
        panel.grid = element_blank())

```

#### Figure 3

The purpose of Figure 3 is to demonstrate divergent selection in male poison dart frog mating call rates over time. This figure shows the different call rates for male poison dart frogs in the Andes region in South America. I chose to use boxplots to visualize the data because it allows the viewer to see both the differences between each group and the distribution of the data within each region.

```{r Divergent Selection w/ Frog Data, message=FALSE}
ggplot(frogs) +
  # make a boxplot geometry comparing call rates by region
  geom_boxplot(aes(x = Region, y = CR, fill = Region)) +
  # set the color scheme to match the other figures
  scale_fill_brewer(palette = "Set1") +
  # set labels for title, caption, and y axis
  labs(title = "Divergent Selection\nin Poison Dart Frogs",
       caption = "Figure 3. Call rates for male frogs in different regions of South America.",
       y = "Call Rate (1/s)") +
  # set theme to black & white to match the other figures
  theme_bw() +
  theme(# set all text font to Bookman
        text = element_text(family = "Bookman"),
        # change sizing and location of title and caption
        plot.title = element_text(hjust = 0.5, size = 15),
        plot.caption = element_text(hjust = 0, size = 10),
        # remove the minor gridlines to make figure simpler
        panel.grid.minor = element_blank())

```

## Discussion

There are a great number of distinct and useful ways of determining where and how selection is occurring in a species. With a foundational understanding of the three types of selection---stabilizing, directional, and divergent---biologists can better find new ways of understanding these processes through data visualization. Innovations in genomics can provide some of this data, and measurable phenotypic characteristics can be used as markers for these types of change. While this study focused on animal species, plant species show the same different types of selection, as evidenced in a number of different plant phenotypes modeling each type of selection below.

![Different types of selection in plant species [@10.2307/26164894]](plant_selection.png)
